main.js

```diff
@@ -45,13 +44,81 @@ function generateNumbers() {
             allNumbers.push(randNumber);
         }
     }
-    while ((arrayCheckEqualsAsc(allNumbers)) || (arrayCheckEqualsDesc(allNumbers))) {
-        allNumbers = shuffleArray(allNumbers);
+
+    return allNumbers;
+}
+
+function generateNegativeNumbers(numbersWanted) {
+    let allNumbers = [];
+
+    for (let i = 0; i < numbersWanted; i++) {
+        let randNumber = Math.floor(Math.random() * 201) - 100;
+
+        if (allNumbers.includes(randNumber)) {
+            i--;
+        } else {
+            allNumbers.push(randNumber);
+        }
     }
 
     return allNumbers;
 }
 
+function generateHardNumbers(numbersWanted) {
+    let randNumber = '';
+    let allNumbers = [];
+    let numberSelect = Math.floor(Math.random() * 10);
+
+    for (let i = 0; i < numbersWanted; i++) {
```

Comment:
this is very confusing, can it be broken down into smaller functions with semantic names to make this clearer